pallets/pools/src/tranche.rs

```diff
@@ -86,14 +72,52 @@ impl<PoolId, TrancheId> TrancheLocator<PoolId, TrancheId> {
 	}
 }
 
-#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Default, TypeInfo)]
-pub struct Tranches<Balance, Rate, Weight>(pub(super) Vec<Tranche<Balance, Rate, Weight>>);
+#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, TypeInfo)]
+pub enum TrancheType<Rate> {
+	Residual,
+	NonResidual {
+		interest_per_sec: Rate,
+		min_risk_buffer: Perquintill,
+	},
+}
 
-#[derive(Clone, Encode, Decode, Eq, PartialEq, RuntimeDebug, Default, TypeInfo)]
-pub struct Tranche<Balance, Rate, Weight> {
-	pub(super) interest_per_sec: Rate,
-	pub(super) min_risk_buffer: Perquintill,
+impl<Rate> TrancheType<Rate>
+where
+	Rate: PartialOrd + PartialEq,
+{
+	/// Compares tranches with the following schema:
+	///
+	/// * (Residual, Residual) => true
```

Comment:
Maybe I misunderstand this code, but this would imply we can have multiple residual tranches. We don't support this properly right now (although we might support it in the future), as we'd need to modify the tranche debt and tranche token price calculations.

Reply:
Yes, this function does support it. But one can not create a pool with multiple residual tranches see: https://github.com/centrifuge/centrifuge-chain/blob/166e155c6f502a4c1aa1de92513b8b0de2684850/pallets/pools/src/lib.rs#L1465-L1473